### ISSUE TITLE
Extra space is added between each word

### ISSUE THREAD

**Issue by malickf** [danmar/simplecpp#111]
I do not understand why some extra space is added in the output of the preprocessed file. 
For instance instead to get : 

```
int main()
{
int c = func(5);
}
```
I obtain : 
```
int main ( )
{
int c = func ( 5 ) ;
}
```
Is this normal ?

(on windows)

**Comment by danmar** [comment]
the space is only a visual thing.

you get 14 tokens:
int
main
(
)
{
int
c
=
func
(
5
)
;
}

when these tokens are written as plaintext (plaintext format is only used for debugging as far as I know) there is a pointless space. I think it would be fine to remove the space. That would not affect Cppcheck etc as that does not use the plaintext debugging output.

**Comment by malickf** [comment]
Ok many thanks.

**Comment by alienself** [comment]
Same issue here.
I actually need the string and don't care about the token. An option to remove the extra space between each word would have been nice.

**Comment by danmar** [comment]
feel free to provide a pull request. The space is added here:

https://github.com/danmar/simplecpp/blob/master/simplecpp.cpp#L527

**Comment by a1880** [comment]
I've also stumbled across the extra space.
It disturbs my usecase to process non-cpp scripts.

To eliminate the extra space, I did the following:

Class Token gets an extra bool field `preceededBySpace` with default value false.
This also has to be set/copied in the copy constructor simplecpp.h, line 111.

The extra space in simplecpp.cpp line 526 is only added, if `tok->preceededBySpace` is true.

In function `readList()` an extra bool variable `preceededBySpace `is added in line 599 of simplecpp.cpp.
Start value is false. It is changed to true in line 681, if a space is found.
In line 850, preceededBySpace is copied to the token field before pushing the Token to the Token list.
Afterwards, `preceededBySpace ` is reset to false.

Sorry, but I haven't practiced yet how to write pull-requests.

**Comment by a1880** [comment]
I ported simplecpp to Windows. Therefore, my version is incompatible with the main version.
Additionally, I extended simplecpp in minor details. It can now take a named output file. The --help output is also more elaborate.

With respect to the column:
What happens if the column changes due to macro expansion?

[simplecpp.zip](https://github.com/danmar/simplecpp/files/11453017/simplecpp.zip)

**Comment by danmar** [comment]
Sounds complex. I would like something clean to apply.

**Comment by a1880** [comment]
I've only changed minor things:

main.cpp
+  added show_version() to show the program version (new option -V)
     Version is also shown as part of help usage output
+  added new feature to write output to named file rather than stdout (2nd positional parameter)
+  added extended help usage output (new options -? and -h)

simplecpp.cpp
+  automatic source formatting in Visual Studio (added a lot of spaces; hard to follow ..)
+  added some casts to reduce compiler warnings
+  added preceededBySpace Token logic

simplecpp.h
+ defined _CRT_SECURE_NO_WARNINGS to reduce compiler warnings
+ included preceededBySpace filed in Token() copy constructor
+ added preceededBySpace as additional bool field in Token class
+ added explanatory comment DUI = Defines Undefines Includepaths for struct SIMPLECPP_LIB DUI

The actual processing should be unchanged. The overall directory structure however, is different.
It conforms to the Visual Studio requirements. Therefore, I cannot simply clone/push between git and my system.
